openapi3/schema.go

```diff
@@ -956,17 +960,22 @@ func (schema *Schema) visitSetOperations(settings *schemaValidationSettings, val
 
 			return e
 		}
+
+		_ = v[matchedOneOfIdx].Value.visitJSON(settings, value)
```

Comment:
Please wrap these operations and the deepcopy inside 
```go
	if settings.asreq || settings.asrep {
```
the same way that the code that mutates `value` does
```go
	if settings.asreq || settings.asrep {
		for propName, propSchema := range schema.Properties {
			if value[propName] == nil {
				if dlft := propSchema.Value.Default; dlft != nil {
					value[propName] = dlft
					if f := settings.defaultsSet; f != nil {
						settings.onceSettingDefaults.Do(f)
					}
				}
			}
		}
	}
```
This way the general case isn't costlier because of this PR.

Reply:
Make sense, fixed.